packages/sdk/src/oracle/services/erc20.ts

```diff
@@ -0,0 +1,43 @@
+import { erc20 } from "../../clients";
+import Multicall2 from "../../multicall2";
+import { BatchReadWithErrors, BatchReadWithErrorsType, Calls } from "../../utils";
+import { BigNumber, Provider } from "../types/ethers";
+
+const batchProps: Calls = [["symbol"], ["name"], ["decimals"], ["totalSupply"]];
+type Props = {
+  symbol: string;
+  name: string;
+  decimals: number;
+  totalSupply: BigNumber;
+};
+export class Erc20 {
+  public contract: erc20.Instance;
+  constructor(protected provider: Provider, protected address: string) {
+    this.contract = erc20.connect(address, provider);
+  }
+  async getProps(): Promise<Props> {
```

Comment:
spent several hours trying to make this call dynamic, ie provide the props you want rather than hard coding it. could not get typescript to pass.

Reply:
Doesn't seem like a huge deal, but it is odd that it was difficult considering it should be nearly identical to batchRead, right?

Reply:
Part of the prolbem is that you are using user supplied prop arguments, which i couldnt figure out how to convert back into a known type to typescripts satisfaction. I dont remember the exact errors though

Reply:
Interesting. I totally agree that this sort of thing is difficult to make work in ts, but I'm surprised it didn't just follow the same pattern as batchRead which takes user-defined args and produces typed outputs. I could be misunderstanding batchRead though.

Reply:
i didnt try to add generics and require the user to type cast the call, like in batch read. not sure why i didnt think of it

Reply:
also i dont know why typescript does not complain about the batch read function. maybe because we arent calling functions  dynamically on the multicall contract, where as in this call we would be using these user passed strings as ways to key into the contract methods dynamically

Reply:
i spent some more time on this, probably not going to keep trying. but i eventually get to the point where typescript needs me to validate the arguments to the call, which is maybe possible but i dont know

![image](https://user-images.githubusercontent.com/4429761/147669677-df6e20b4-170f-4c42-be69-5776fc10eb85.png)
![image](https://user-images.githubusercontent.com/4429761/147669699-2c63f957-e641-43c2-8286-0e4292b1794d.png)